packages/ms-authenticator/src/authenticators/MsAuthenticator.ts

```diff
@@ -0,0 +1,85 @@
+import { ConfidentialClientApplication, LogLevel, PublicClientApplication, UsernamePasswordRequest } from '@azure/msal-node'
+import { IMsAuthenticationClientCredentialArgs, IMsAuthenticationUsernamePasswordArgs } from '../index'
+
+import { fetch } from 'cross-fetch'
+
+/**
+ * necessary fields are:
+ *   azClientId: clientId of the application you're trying to login
+ *   azClientSecret: secret of the application you're trying to login
+ *   azTenantId: your MS Azure tenantId
+ *   credentialManifest: address of your credential manifest. usually in following format:
+ *    https://beta.eu.did.msidentity.com/v1.0/<tenant_id>/verifiableCredential/contracts/<verifiable_credential_schema>
+ * @param authenticationArgs
+ * @constructor
+ */
+export async function ClientCredentialAuthenticator(authenticationArgs: IMsAuthenticationClientCredentialArgs): Promise<string> {
+  const msalConfig = {
+    auth: {
+      clientId: authenticationArgs.azClientId,
+      authority: authenticationArgs.authority ? authenticationArgs.authority : 'https://login.microsoftonline.com/' + authenticationArgs.azTenantId,
+      clientSecret: authenticationArgs.azClientSecret,
+    },
+    system: {
+      loggerOptions: {
+        piiLoggingEnabled: false,
+        logLevel: LogLevel.Verbose,
+      }
+    }
+  }
+
+  const cca = new ConfidentialClientApplication(msalConfig)
+  const msalClientCredentialRequest = {
+    scopes: authenticationArgs.scopes ? authenticationArgs.scopes : ['3db474b9-6a0c-4840-96ac-1fceb342124f/.default'],
```

Comment:
Can we make a const out of this with a meaningful name?

Reply:
changed